ring/func/sync.py

```diff
@@ -205,11 +206,82 @@ def touch_value(self, key, expire):
+        assert isinstance(backend, type({})), 'backend has to be dict-like'
+        self._backend = backend
+        self._target_size = round(target_size)
+        self._expire_f = expire_f
+        self._mutex = threading.Lock()
+
+    def run(self):
+        class WorkThread(threading.Thread):
+            DEBUG = False
+
+            def __init__(self, outer_instance):
+                threading.Thread.__init__(self)
+                self._backend = outer_instance._backend
+                self._target_size = outer_instance._target_size
+                self._expire_f = outer_instance._expire_f
+                self._mutex = outer_instance._mutex
+
+            def strategy_with_expire(self):
+                MAX_EXPIRE_RETRY_COUNT = 4
+                now = time.time()
+                retry_count = 0
+                keys = list(self._backend.keys())
+                while (len(self._backend) > self._target_size) and (retry_count < MAX_EXPIRE_RETRY_COUNT):
+                    k = keys.pop(random.randrange(len(keys)))
+                    val = self._backend.get(k, None)
+                    if val is None:
+                        continue
+                    expire = self._expire_f(val)
+                    if expire < now:
+                        self._backend.pop(k, None)
+                        self.print_d('{} removed from strategy_with_expire => size {}'.format(k, len(self._backend)))
+                    else:
+                        retry_count += 1
+                    if len(keys) == 0:
+                        keys = list(self._backend.keys())
+
+            def strategy_with_force(self):
+                keys = list(self._backend.keys())
+                while (len(self._backend) > self._target_size) and len(keys) > 0:
+                    k = keys.pop(random.randrange(len(keys)))
+                    self._backend.pop(k, None)
+                    self.print_d('{} removed from strategy_with_force => size {}'.format(k, len(self._backend)))
+                    if len(keys) == 0:
+                        keys = list(self._backend.keys())
+
+            def run(self):
+                try:
+                    self.print_d('gc started in size:{}'.format(len(self._backend)))
+                    if self._expire_f is not None:
+                        self.strategy_with_expire()
+                    self.strategy_with_force()
+                    self.print_d('gc ended in size:{}'.format(len(self._backend)))
+                finally:
+                    self._mutex.release()
+
+
+            def print_d(self, str_):
+                if self.DEBUG:
+                    print(str_)
+
+        self._mutex.acquire()
+        if (len(self._backend) > self._target_size):
+            WorkThread(self).start()
+        else:
+            self._mutex.release()
 
 class ExpirableDictStorage(fbase.CommonMixinStorage, fbase.StorageMixin):
-
+    maxsize = 128
```

Comment:
This value should not be bound to the class

Reply:
Again, I can't 100% follow your thoughts with `maxsize`. 
My guess is you're worrying that this value can be easily accessed & modified and my understanding is `ExpirableDictStorage` & `PersistentDictStorage` is our implement for default calling so we can control this value.